src/insights.ts

```diff
@@ -134,54 +111,80 @@ class AlgoliaInsights {
     });
   }
 
-  public clickedObjectIDs(event: ObjectIDsEvent) {
+  clickedObjectIDs(event: ObjectIDsEvent) {
     this.sendEvent({
       eventType: 'click',
       ...event,
     });
   }
 
-  public clickedFilters(event: FiltersEvent) {
+  clickedFilters(event: FiltersEvent) {
     this.sendEvent({
       eventType: 'click',
       ...event,
     });
   }
 
-  public convertedObjectIDsAfterSearch(event: ObjectIDsAfterSearchEvent) {
+  convertedObjectIDsAfterSearch(event: ObjectIDsAfterSearchEvent) {
     this.sendEvent({
       eventType: 'conversion',
       ...event,
     });
   }
 
-  public convertedObjectIDs(event: ObjectIDsEvent) {
+  convertedObjectIDs(event: ObjectIDsEvent) {
     this.sendEvent({
       eventType: 'conversion',
       ...event,
     });
   }
 
-  public convertedFilters(event: FiltersEvent) {
+  convertedFilters(event: FiltersEvent) {
     this.sendEvent({
       eventType: 'conversion',
       ...event,
     });
   }
 
-  public viewedObjectIDs(event: ObjectIDsEvent) {
+  viewedObjectIDs(event: ObjectIDsEvent) {
     this.sendEvent({
       eventType: 'view',
       ...event,
     });
   }
 
-  public viewedFilters(event: FiltersEvent) {
+  viewedFilters(event: FiltersEvent) {
     this.sendEvent({
       eventType: 'view',
       ...event,
     });
   }
+
+  private sendEvent(event: Omit<InsightsApiEvent, 'timestamp' | 'userToken'>) {
+    if (!this.beacon || !this.userToken) {
+      throw new Error(
+        "Before calling any other method, you need to initialize the library by calling the 'init' function with appId and apiKey parameters"
+      );
+    }
+
+    const userToken = this.userToken.getUserToken();
+    if (!userToken) {
+      throw new Error('userToken required to send event');
+    }
+
+    this.beacon.send({
+      userToken,
+      timestamp: Date.now(),
+
+      ...event,
+    });
+  }
+
+  private addAlgoliaAgent(agent: string) {
+    this.beacon?.addAlgoliaAgent(agent);
+  }
+
+  [k: string]: any;
```

Comment:
This is a strange part of the interface. What is it used for?

Reply:
This is a hangover from my first version of this, can be removed.